src/client/components/input/drag-and-drop-image.js

```diff
@@ -1,11 +1,25 @@
 const React = require('react');
 const Input = require('react-bootstrap').Input;
 
+/**
+* This class is derived from the React Component base class to render the
+* drag and drop image functionality.
+*/
 class DragAndDropImage extends React.Component {
+	/**
+	* Binds the class methods to their respective data.
+	* @constructor
+	*/
 	constructor() {
 		super();
 		this.handleDragStart = this.handleDragStart.bind(this);
 	}
+	/**
+	* Binds the data of the achievement badge on drag start to the data of the
```

Comment:
Where does this data end up? Take a look at the other drag and drop components to see if you can find out, and then add some relevant information here. I'd also reword this comment to talk about transfer of data from the component properties to the event.